### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,2 +1,49 @@
 This package provides an ExplainCollection class
-that allows common MongoDB commands to be explained
\ No newline at end of file
+that allows common MongoDB commands to be explained
+
+Tutorial
+########
+The intended use case for this package is to debug ``pymongo`` commands.
+This can be done by simply swapping out ``Collection`` for ``ExplainCollection``,
+which has the same functions but will run explain on them *instead* of executing them.
```
Comment (ShaneHarvey) [inline README.rst:8]: nit: functions->methods

File: README.rst [hunk 2]
```diff
@@ -1,2 +1,49 @@
...
+The first step is to create a ``MongoClient`` instance::
+
+    client = MongoClient(serverSelectionTimeoutMS=1000, event_listeners=[logger])
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply README.rst:11]: Done.

File: README.rst (hunk 1 above)
Comment (juliusgeo) [reply README.rst:8]: Done.

File: README.rst [hunk 3]
```diff
@@ -1,2 +1,49 @@
 This package provides an ExplainCollection class
-that allows common MongoDB commands to be explained
\ No newline at end of file
+that allows common MongoDB commands to be explained
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply README.rst:2]: Done.

File: README.rst [hunk 4]
```diff
@@ -1,2 +1,49 @@
...
+        '$clusterTime':{
+                'signature': {'keyId': 0,
+                'hash': b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'},
+                'clusterTime': Timestamp(1594666552, 1)
+         },
+        'queryPlanner': {
+            'plannerVersion': 1,
+            'rejectedPlans': [],
+            'indexFilterSet': False,
+            'namespace': 'db.products', 'winningPlan': {'stage': 'EOF'}
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply README.rst:43]: Done.

File: README.rst [hunk 5]
```diff
@@ -1,2 +1,54 @@
 This package provides an ExplainCollection class
```
Comment (prashantmital) [inline README.rst:1]: Let's quote the names of all classes - ``ExplainCollection``

File: README.rst [hunk 6]
```diff
@@ -1,2 +1,54 @@
...
-that allows common MongoDB commands to be explained
\ No newline at end of file
+that allows that allows PyMongo's Collection methods to be explained.
```
Comment (prashantmital) [inline README.rst:2]: Can you link "explained" to the MongoDB public documentation of the `explain` command?

File: README.rst [hunk 7]
```diff
@@ -1,2 +1,54 @@
...
+
+
+Tutorial
+########
+The intended use case for this package is to debug ``pymongo`` commands.
```
Comment (prashantmital) [inline README.rst:7]: > ...to debug...

I am not sure that is correct. I think the use case is to ``explain`` operations run using PyMongo

File: README.rst [hunk 8]
```diff
@@ -1,2 +1,54 @@
...
+This can be done by simply swapping out ``Collection`` for ``ExplainCollection``,
+which has the same methods but will run explain on them *instead* of executing them.
```
Comment (prashantmital) [inline README.rst:9]: Nit: suggest boldface **instead**

File: README.rst [hunk 9]
```diff
@@ -1,2 +1,54 @@
...
+The first step is to create a ``MongoClient`` instance::
+
+    client = MongoClient('mongodb://localhost:27017/')
```
Comment (prashantmital) [inline README.rst:12]: I would omit this step and directly with the next step. Client creation is something users must already be doing and is not something they need to do specially when wanting to use `pymongoexplain`.

File: README.rst [hunk 10]
```diff
@@ -1,2 +1,54 @@
...
+
+Once this is done, you can simply get a collection instance, and then wrap it in the ``ExplainCollection`` class.::
```
Comment (prashantmital) [inline README.rst:14]: Suggest rewording: 

To explain operations in your application, you must first create an `ExplainCollection` object. Creating one from a vanilla PyMongo collection is straightforward:

<code example>

File: README.rst [hunk 11]
```diff
@@ -1,2 +1,54 @@
...
+
+    collection = client.db.products
+    explain = ExplainCollection(collection)
+
+Now you are ready to run some commands. Remember that these commands will not be executed, they will simply have explain
```
Comment (prashantmital) [inline README.rst:19]: *now you are ready to **explain** some commands

File: README.rst (hunk 5 above)
Comment (juliusgeo) [reply README.rst:1]: Done.

File: README.rst (hunk 6 above)
Comment (juliusgeo) [reply README.rst:2]: Done.

File: README.rst (hunk 7 above)
Comment (juliusgeo) [reply README.rst:7]: Done.

File: README.rst (hunk 8 above)
Comment (juliusgeo) [reply README.rst:9]: Done.

File: README.rst (hunk 9 above)
Comment (juliusgeo) [reply README.rst:12]: Done.

File: README.rst (hunk 11 above)
Comment (juliusgeo) [reply README.rst:19]: done.

File: README.rst [hunk 12]
```diff
@@ -1,2 +1,68 @@
...
\ No newline at end of file
+This package provides an ``ExplainCollection`` class
+that allows that allows PyMongo's Collection methods to be explained_
+
+.. _explained: https://docs.mongodb.com/master/reference/command/explain/#dbcmd.explain.
+
+
+Tutorial
+########
+The intended use case for this package is to allow ``pymongo`` commands to be easily explained_.
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply README.rst:9]: Done.

File: README.rst [hunk 13]
```diff
@@ -1,2 +1,68 @@
...
+This can be done by simply swapping out ``Collection`` for ``ExplainCollection``,
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply README.rst:10]: Done.

File: README.rst [hunk 14]
```diff
@@ -1,2 +1,68 @@
...
+which has the same methods but will run explain on them **instead** of executing them.
+The first step is to create a ``MongoClient`` instance, then, you can simply get a collection instance, and then wrap it in the ``ExplainCollection`` class.::
+
+    collection = client.db.products
+    explain = ExplainCollection(collection)
+
+Now you are ready to explain some commands. Remember that these commands will not be executed, they will simply have explain
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply README.rst:17]: Done.

File: README.rst [hunk 15]
```diff
@@ -1,2 +1,68 @@
...
+run on them.::
+
+    res = explain.update_one({"quantity": 1057, "category": "apparel"}, {"$set": {"reorder": True}})
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply README.rst:20]: Done.

File: README.rst [hunk 16]
```diff
@@ -1,2 +1,68 @@
...
+
+The value of ``res`` will be whatever the output of explain for that ``update_one`` command is: ::
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply README.rst:22]: Done.

File: README.rst [hunk 17]
```diff
@@ -1,2 +1,68 @@
...
+                            {'$and': [
+                                {'category': {'$eq': 'apparel'}},
+                      {'quantity': {'$eq': 1057}}]},
+                       'stage': 'COLLSCAN', 'direction': 'forward'},
+                       'stage': 'UPDATE'}, 'planCacheKey': 'CD8F6D8F',
+                      'rejectedPlans': []}, 'ok': 1.0,
+                      'operationTime': Timestamp(1595603051, 3)}
+
+
+This diagnostic information should hopefully help you understand what the problem is with your commands. Because
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply README.rst:50]: Done.

File: README.rst [hunk 18]
```diff
@@ -1,2 +1,68 @@
...
+``ExplainCollection`` implements all the methods of ``Collection``, you can simply replace instances of collection with
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply README.rst:51]: Done.

File: README.rst [hunk 19]
```diff
@@ -1,2 +1,68 @@
...
+an ``ExplainCollection`` to get this information.
+
+CLI Tool
+########
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply README.rst:55]: Done.

File: README.rst [hunk 20]
```diff
@@ -1,2 +1,68 @@
...
+
+You can also run explain on all commands within a Python script using our CLI tool.
+Given a script that contains ``pymongo`` commands within it, you can simply run: ::
+
+    python3 -m pymongoexplain ../<pymongoscript>.py
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply README.rst:60]: Done.

File: README.rst [hunk 21]
```diff
@@ -1,2 +1,68 @@
...
+
+This will print out the explain output for every single command
+within that script, in addition to running the script itself.
+
+If you have arguments that need to be supplied, you can simply append those
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply README.rst:65]: Done.

File: README.rst [hunk 22]
```diff
@@ -1,2 +1,68 @@
...
+that allows that allows PyMongo's Collection methods to be explained_
+
+.. _explained: https://docs.mongodb.com/master/reference/command/explain/#dbcmd.explain.
+
+
+Tutorial
+########
+
+PyMongo operations in existing application code can be explained by swapping Collection objects with ExplainCollection
+objects. The ExplainCollection class provides all CRUD API methods provided by PyMongo's Collection,
```
Comment (prashantmital) [inline README.rst:11]: Please use backticks when mentioning classnames: ``ExplainCollection``

Comment (prashantmital) [reply README.rst:11]: Also, PyMongo's ``Collection``

File: README.rst [hunk 23]
```diff
@@ -1,2 +1,68 @@
...
+but using this class to run operations runs explain on them, instead of executing them.
+
+To run explain on a command, first instantiate an ExplainCollection from the Collection object originally used to run the command::
```
Comment (prashantmital) [inline README.rst:14]: Same comment about using backticks for classnames

**Comment by prashantmital** [comment]
Note that the README still needs to be populated with other project information. See #19
